File: Linked_List_Cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */

 // Imports
import java.util.List;
import java.util.ArrayList;

public class Solution {
    public boolean hasCycle(ListNode head) {
        List<ListNode> list = new ArrayList<ListNode>();
        ListNode cursor = head;

        while(cursor != null){
            if(list.contains(cursor)){
                return true;
            }

            list.add(cursor); 
            cursor = cursor.next;
        }
        return false;
    }
}

File: Linked_List_Cycle.py
# My second "Two-Pointer" solution to this problem
class Solution(object):
    def hasCycle(self, head):
        cursor = head
        twoStepCursor = head
        
        while twoStepCursor and twoStepCursor.next:
            cursor = cursor.next
            twoStepCursor = twoStepCursor.next.next
            
            if cursor == twoStepCursor:
                return True

        return False
